llvm/include/llvm/CodeGen/SelectionDAGNodes.h

```diff
@@ -837,29 +835,72 @@ END_TWO_BYTE_PACK()
     }
   };
 
+  class user_iterator {
+    friend class SDNode;
+    use_iterator UI;
```

Comment:
This might be a silly question but can you implement this as `class user_iterator : use_iterator` and just override `value_type` and `operator*`?